presto-main/src/main/java/com/facebook/presto/sql/SqlPath.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.String.format;
+
+public class SqlPath
+{
+    private List<SqlPathElement> parsedPath;
+    private String rawPath;
+
+    @JsonCreator
+    public SqlPath(@JsonProperty("rawPath") String path)
+    {
+        this.rawPath = path;
+    }
+
+    @JsonProperty
+    public String getRawPath()
+    {
+        return rawPath;
+    }
+
+    public List<SqlPathElement> getParsedPath()
```

Comment:
What do you think about naming `parsedPath` as `path`, so this method becomes `getPath`, which I think will simplify the name a bit? We can also rename `rawPath` as `pathString`.

Reply:
I wanted to avoid this because it creates situations in which a line of code could look like: 
```session.getPath().getPath();```
The choice was to reserve the name ```getPath()``` for retrieving the SqlPath object, and that from there, the option was to get either the parsed version or the raw version.